### PR TITLE
Pin paramiko to 2.4.2, update requirements

### PR DESCRIPTION
**Pull request by fredreichbier**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+cffi==1.12.3
+configobj==5.0.6
+cryptography==2.6.1
+enum34==1.1.6
 funcsigs==1.0.2
-paramiko==1.17.6
+ipaddress==1.0.22
+mock==2.0.0
+netaddr==0.7.19
+nose==1.3.7
```
Comment (plettich) [inline requirements.txt]: I think we can get rid of `nose`...
I find it more clear to have an additional `requirements.txt` file in `test` which lists all the packages required to run the tests. Thus we keep the base `requirements.txt` clean.

File: requirements.txt [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
-argparse==1.2.1
+asn1crypto==0.24.0
+bcrypt==3.1.6
+cffi==1.12.3
+configobj==5.0.6
+cryptography==2.6.1
+enum34==1.1.6
 funcsigs==1.0.2
-paramiko==1.17.6
+ipaddress==1.0.22
```
Comment (plettich) [inline requirements.txt]: `enum34` and `ipaddress` are only needed with python 2.7. Should we make these requirements conditional?

File: requirements.txt (hunk 1 above)
Comment (fredreichbier) [reply requirements.txt]: Good point, I'll remove it.

File: requirements.txt (hunk 2 above)
Comment (fredreichbier) [reply requirements.txt]: Sounds good, will do!

Comment (fredreichbier) [reply requirements.txt]: done in 37e544521ddd9d821148e53455054bbbd368dcad

File: requirements.txt (hunk 1 above)
Comment (fredreichbier) [reply requirements.txt]: done in 414e922, same for ``mock`` :)

File: test/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+-r ../requirements.txt
+mock==2.0.0
+nose==1.3.7
```
Comment (plettich) [inline test/requirements.txt]: actually i would prefer `pytest` since i already use it in the travis configuration :-)

Comment (fredreichbier) [reply test/requirements.txt]: Fine by me :) Fixed this in be5d460cf48a4457ae11b5629ba0c83b6d6b602a

File: requirements.txt (hunk 1 above)
Comment (cornelinux) [reply requirements.txt]: lgtm

**Pull request merged**